src/overlays/actors/ovl_Bg_Ice_Shutter/z_bg_ice_shutter.c

```diff
@@ -27,19 +31,104 @@ const ActorInit Bg_Ice_Shutter_InitVars = {
     (ActorFunc)BgIceShutter_Update,
     (ActorFunc)BgIceShutter_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Shutter/func_80891AC0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Shutter/BgIceShutter_Init.s")
+extern UNK_TYPE D_06002854;
+extern Gfx D_06002740[];
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+void func_80891AC0(BgIceShutter* thisx) {
+    f32 sp24;
+
+    sp24 = Math_Sins(thisx->dyna.actor.shape.rot.x) * thisx->dyna.actor.velocity.y;
+    thisx->dyna.actor.posRot.pos.y =
+        (Math_Coss(thisx->dyna.actor.shape.rot.x) * thisx->dyna.actor.velocity.y) + thisx->dyna.actor.initPosRot.pos.y;
+    thisx->dyna.actor.posRot.pos.x =
+        (Math_Sins(thisx->dyna.actor.shape.rot.y) * sp24) + thisx->dyna.actor.initPosRot.pos.x;
+    thisx->dyna.actor.posRot.pos.z =
+        (Math_Coss(thisx->dyna.actor.shape.rot.y) * sp24) + thisx->dyna.actor.initPosRot.pos.z;
+}
+
+void BgIceShutter_Init(Actor* thisx, GlobalContext* globalCtx) {
+    f32 pad;
```

Comment:
?32 pads are usually s32 in the codebase

Reply:
actually can you replace this pad with a `BgIceShutter* this = THIS;` and use this instead of thisx/THIS in the rest of the function?